orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/ExpressionAware.kt

```diff
@@ -130,7 +136,9 @@ interface ExpressionAware {
 
   private fun StageContext.augmentContext(execution: Execution): StageContext =
```

Comment:
ideally, we opt out of this method and use `ContextParameterProcessor.buildExecutionContext` but that doesn't generate `StageContext` preventing us from doing recursive lookups... An investigation for another day

Reply:
worth adding as a todo  comment so we dont lose context after this pr is forgotten?

Reply:
Hi friends! Sorry to surface bad memories, but I came across this PR while investigating [this bug](https://github.com/spinnaker/spinnaker/issues/4932), which reports that we are failing to evaluate SpEL in Kubernetes manifests downloaded as artifacts. Since we are generating the execution context for the [mid-task SpEL evaluation](https://github.com/spinnaker/orca/blob/master/orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/ManifestEvaluator.java#L132) using `ContextParameterProcessor.buildExecutionContext`, which @marchello2000 correctly points out does not return `StageContext`, the evaluation fails. I've verified that changing the return type of that method from `Map<String, Object>` to `StageContext` resolves the issue, but wanted to check if there had been additional discussion offline about this before I submitted a PR. Thanks in advance!

Love,
Sleepless in SpEL-hell